Updated Exodus decayer

otonvd opened the pull request:
Changes as well in alipythia to allow new decays to be handled by Exodus

alibuild commented:
95c79271c053e5466ef045464409af61eec54edf: approval required: 1 of @rbertens (Redmer Alexander Bertens), @qgp (Jochen Klein); 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/433) and comment at the bottom of the page.**_

dberzano commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -51,11 +51,13 @@ AliDecayerExodus::AliDecayerExodus():
     fEPMassPion(0),
     fEPMassEta(0),
     fEPMassEtaPrime(0),
+fEPMassEtaPrime_toOmega(0),
```
Fix indentation

otonvd replied:
Hi Dario, should  I close the pull request and make a new one?

dberzano replied:
If you keep pushing to your current branch, the PR will be updated. See [here](http://alisw.github.io/git-tutorial/#amend-an-open-pull-request).

dberzano commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -51,11 +51,13 @@ AliDecayerExodus::AliDecayerExodus():
     fEPMassOmegaDalitz(0),
     fEPMassPhi(0),
     fEPMassPhiDalitz(0),
+fEPMassPhiDalitz_toPi0(0),
```
Same here

dberzano commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -113,54 +115,68 @@ void AliDecayerExodus::Init()
     maxeta      = etamass - omassgamma;
     maxomega    = omegamass - pionmass;
     maxetaprime = etaprimemass - omassgamma;
+maxetaprime_toOmega = etaprimemass - omegamass;
```
Indentation again

alibuild commented:
e5cf0fac1dc215bfe64dcdda387e20d0ce5f1894: approval required: 1 of @rbertens (Redmer Alexander Bertens), @qgp (Jochen Klein); 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/433) and comment at the bottom of the page.**_

rbertens commented:
+test

alibuild commented:
e5cf0fac1dc215bfe64dcdda387e20d0ce5f1894: approval required: 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/433) and comment at the bottom of the page.**_

ktf commented:
@dberzano is there anything stuck for the PR checking? This seems to be stalling...

ktf commented:
+test

alibuild commented:
e5cf0fac1dc215bfe64dcdda387e20d0ce5f1894: testing approved: will not be automatically merged; starting testing. If testing succeeds, merging will require further approval from 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

alibuild commented:
e5cf0fac1dc215bfe64dcdda387e20d0ce5f1894: tests OK, approval required for merging: 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to merge. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/433) and comment at the bottom of the page.**_

alibuild commented:
Error while checking build/AliRoot/release for e5cf0fac1dc215bfe64dcdda387e20d0ce5f1894:
```

```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/433/e5cf0fac1dc215bfe64dcdda387e20d0ce5f1894/fullLog.txt).

alibuild commented:
d183d8d41e372b052bbd8065b1effa1094719aeb: approval required: 1 of @rbertens (Redmer Alexander Bertens), @qgp (Jochen Klein); 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/433) and comment at the bottom of the page.**_

otonvd commented:
Dear all, since this is one of my first commits using github, I am now a bit lost. What's the status of this commit? Is there any action required from my side?

dberzano commented:
+test

As the last comment says, one of the mentioned persons must approve the commit. Approval is required every time you modify the PR (not just when you open it). Some expert on the given matter should give the ultimate approval if this PR compiles (I have approved test runs without automatic merging).

alibuild commented:
d183d8d41e372b052bbd8065b1effa1094719aeb: testing approved: will not be automatically merged; starting testing. If testing succeeds, merging will require further approval from 1 of @rbertens (Redmer Alexander Bertens), @qgp (Jochen Klein); 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

alibuild commented:
d183d8d41e372b052bbd8065b1effa1094719aeb: tests OK, approval required for merging: 1 of @rbertens (Redmer Alexander Bertens), @qgp (Jochen Klein); 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to merge. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/433) and comment at the bottom of the page.**_

alibuild commented:
Error while checking build/AliRoot/release for d183d8d41e372b052bbd8065b1effa1094719aeb:
```

```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/433/d183d8d41e372b052bbd8065b1effa1094719aeb/fullLog.txt).

dberzano commented:
I believe there is something wrong with the AliRoot tester. Let me restart it. The error will clear out automatically, ignore it for now.

alibuild commented:
Error while checking build/AliRoot/release for d183d8d41e372b052bbd8065b1effa1094719aeb:
```

```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/433/d183d8d41e372b052bbd8065b1effa1094719aeb/fullLog.txt).

alibuild commented:
Error while checking build/AliRoot/release for d183d8d41e372b052bbd8065b1effa1094719aeb:
```

```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/433/d183d8d41e372b052bbd8065b1effa1094719aeb/fullLog.txt).

otonvd commented:
Unfortunately I cannot see the log...

dberzano commented:
@otonvd the infrastructure has temporary problems that do not depend on us (but CERN IT). The message is misleading, but please ignore it. The PR will become green as soon as the problem is fixed.

alibuild commented:
dcb43b06a9e91d77f54f161979cfee5f629305c6: approval required: 1 of @rbertens (Redmer Alexander Bertens), @qgp (Jochen Klein); 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/433) and comment at the bottom of the page.**_

otonvd commented:
I see that I create a pull request everytime I force push (the only way I can actually push). I don't know how to solve it!

rbertens commented:
Hi @otonvd , would be good if @dberzano  could confirm, but I think this happens because as long as this current PR is open, every time you push, the changes you've made are added as an 'addendum' to this open PR.

miweberSMI commented:
Hi all,

what is the status of this PR? I thought it shoudl be approved by now?

Thanks and cheers,
Michael

dberzano commented:
> I see that I create a pull request everytime I force push (the only way I can actually push). I don't know how to solve it!

There's nothing to solve there. It is on purpose, @rbertens is right. When you force-push, you change hash. New hashes are new commits (we identify commits by hash). When a new hash is pushed, approval has to be requested for that new hash. We cannot reuse the approval for another hash, the reason being: imagine if you get the approval for some content, and then you change it, and we keep the former approval...

Besides: approval from PWGMM is required before I approve (I am admin, if I approve my part, it gets merged right away, so PWGMM must `+1` first). This is not a formality, PWGMM should really check what you have done and see if it's OK or not, so please wait for @rbertens or @qgp first.

dberzano commented:
+test

alibuild commented:
dcb43b06a9e91d77f54f161979cfee5f629305c6: testing approved: will not be automatically merged; starting testing. If testing succeeds, merging will require further approval from 1 of @rbertens (Redmer Alexander Bertens), @qgp (Jochen Klein); 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

alibuild commented:
dcb43b06a9e91d77f54f161979cfee5f629305c6: tests OK, approval required for merging: 1 of @rbertens (Redmer Alexander Bertens), @qgp (Jochen Klein); 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @chiarazampolli (Chiara Zampolli), @qgp (Jochen Klein), @dberzano (Dario Berzano), @sawenzel (Sandro Christian Wenzel), @ktf (Giulio Eulisse), @shahor02 (Ruben Shahoyan)

_Comment with `+1` to merge. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/433) and comment at the bottom of the page.**_

rbertens commented:
Hi @miweberSMI , @otonvd , all,

The changes to EVGEN/* are fine from my side, but since this PR also includes some changes to PYTHIA6 (which are also outside of what I can '+1') it would be good to have the input from @qgp . 

More generally, as this changes some fundamental code used by multiple generators, it would be very useful if we could have some additional information, e.g.: was this code validated? is there a JIRA ticket connected? was it discussed in a broader scope? 

Thanks and apologies for the delay,

Redmer

alibuild commented:
Error while checking build/AliRoot/release for dcb43b06a9e91d77f54f161979cfee5f629305c6:
```
sw/BUILD/AliRoot-latest/log
      Start 82: load_library_THepMCParser
      Start 80: load_library_EvtGenExternal
72/93 Test #80: load_library_EvtGenExternal ......   Passed    0.15 sec
73/93 Test #82: load_library_THepMCParser ........   Passed    0.25 sec
      Start 85: load_library_Tauola
      Start 15: load_library_pythia6_4_21
74/93 Test #71: load_library_MONITOR .............   Passed    0.36 sec
75/93 Test  #2: load_library_ANALYSISaliceBase ...   Passed    0.36 sec
      Start  8: load_library_HERWIG
      Start 21: load_library_pythia8210dev
76/93 Test #15: load_library_pythia6_4_21 ........   Passed    0.22 sec
77/93 Test #21: load_library_pythia8210dev .......   Passed    0.13 sec
78/93 Test #85: load_library_Tauola ..............   Passed    0.34 sec
      Start 83: load_library_HepMC
      Start 12: load_library_lhapdf_5_9_1
      Start 81: load_library_EvtGen
79/93 Test  #8: load_library_HERWIG ..............   Passed    0.24 sec
80/93 Test #12: load_library_lhapdf_5_9_1 ........   Passed    0.21 sec
      Start 17: load_library_pythia6_4_28
      Start 75: load_library_StarLight
81/93 Test #83: load_library_HepMC ...............   Passed    0.32 sec
82/93 Test #81: load_library_EvtGen ..............   Passed    0.31 sec
83/93 Test #75: load_library_StarLight ...........   Passed    0.17 sec
      Start  9: load_library_HIJING
      Start 13: load_library_pythia6
      Start 20: load_library_pythia8210
84/93 Test #17: load_library_pythia6_4_28 ........   Passed    0.17 sec
85/93 Test #20: load_library_pythia8210 ..........   Passed    0.19 sec
86/93 Test #13: load_library_pythia6 .............   Passed    0.19 sec
      Start 84: load_library_Photos
      Start 11: load_library_lhapdf
      Start 22: load_library_AMPT
87/93 Test #84: load_library_Photos ..............   Passed    0.16 sec
88/93 Test  #9: load_library_HIJING ..............   Passed    0.35 sec
89/93 Test #11: load_library_lhapdf ..............   Passed    0.15 sec
      Start  6: load_library_dime
      Start 10: load_library_lhapdfbase
      Start 16: load_library_pythia6_4_25
90/93 Test #22: load_library_AMPT ................   Passed    0.16 sec
91/93 Test  #6: load_library_dime ................   Passed    0.15 sec
92/93 Test #16: load_library_pythia6_4_25 ........   Passed    0.20 sec
93/93 Test #10: load_library_lhapdfbase ..........   Passed    0.20 sec

100% tests passed, 0 tests failed out of 93

Total Test time (real) =  15.15 sec
+ rsync -a /mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/0/0/test/ /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/AliRoot/0-1/test
+ [[ RELWITHDEBINFO == COVERAGE ]]
+ mkdir -p /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/AliRoot/0-1/etc/modulefiles
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/433/dcb43b06a9e91d77f54f161979cfee5f629305c6/fullLog.txt).

alibuild commented:
Error while checking build/aliroot/macos for dcb43b06a9e91d77f54f161979cfee5f629305c6:
```

```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/433/dcb43b06a9e91d77f54f161979cfee5f629305c6/fullLog.txt).

alibuild commented:
Error while checking build/AliRoot/macos for dcb43b06a9e91d77f54f161979cfee5f629305c6:
```

```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/433/dcb43b06a9e91d77f54f161979cfee5f629305c6/fullLog.txt).

ktf commented:
@dberzano any reason for not merging this?

dberzano commented:
Yes, waiting for @rbertens or @qgp approval.

rbertens commented:
HI all, I gave a comment already quite a wihle ago:

> The changes to EVGEN/* are fine from my side, but since this PR also includes some changes to PYTHIA6 (which are also outside of what I can '+1') it would be good to have the input from @qgp .

> More generally, as this changes some fundamental code used by multiple generators, it would be very useful if we could have some additional information, e.g.: was this code validated? is there a JIRA ticket connected? was it discussed in a broader scope?

> Thanks and apologies for the delay,

> Redmer

miweberSMI commented:
Hello @dberzano, all,

I am wondering, what happened to this pull request?

thanks and cheers,
Michael

dberzano commented:
@miweberSMI we are waiting for a feedback from @qgp, who's been on vacation for quite some time. @rbertens said that it would be good to have a feedback from him before approving.

miweberSMI commented:
OK, thanks for clarification.

miweberSMI commented:
Hi @qgp,

are there any opinion from your side on this pull reuqest? We would like to have it for another porting request for LHC15o productions...

Thanks and cheers,
Michael

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -346,6 +384,7 @@ Double_t AliDecayerExodus::Lorentz(Float_t mass, Double_t vmass, Double_t vwidth
 
 }
 
+//void AliDecayerExodus::Decay(Int_t idpart, TLorentzVector* pparent, Int_t idpartner)
```
should be removed

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -804,7 +860,8 @@ void AliDecayerExodus::Decay(TClonesArray *array)
 
   TLorentzVector Pizero(part->Px(), part->Py(), part->Pz(), part->Energy());
-  Decay(111, &Pizero);
+  //  Decay(111, &Pizero, dp3[0]->GetPdgCode());
```
remove

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -826,7 +883,8 @@ void AliDecayerExodus::Decay(TClonesArray *array)
 
   TLorentzVector Eta(part->Px(), part->Py(), part->Pz(), part->Energy());
-  Decay(221, &Eta);
+  //Decay(221, &Eta, dp3[0]->GetPdgCode());
```
remove

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -847,6 +905,7 @@ void AliDecayerExodus::Decay(TClonesArray *array)
   if((dp2[0]->GetPdgCode() != 11) && ((TMath::Abs(dp2[1]->GetPdgCode()) != 11)))   continue;
 
   TLorentzVector Rho(part->Px(), part->Py(), part->Pz(), part->Energy());
+  //Decay(113, &Rho, 0);
```
remove

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -868,7 +927,8 @@ void AliDecayerExodus::Decay(TClonesArray *array)
 
   TLorentzVector Omegadalitz(part->Px(), part->Py(), part->Pz(), part->Energy());
-  Decay(223, &Omegadalitz);
+  //Decay(223, &Omegadalitz, dp3[0]->GetPdgCode());
```
remove

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -897,10 +958,11 @@ void AliDecayerExodus::Decay(TClonesArray *array)
 
   TLorentzVector Etaprime(part->Px(), part->Py(), part->Pz(), part->Energy());
-  Decay(331, &Etaprime);
+  //Decay(331, &Etaprime, dp3[0]->GetPdgCode());
```
remove

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -918,7 +980,8 @@ void AliDecayerExodus::Decay(TClonesArray *array)
 
   TLorentzVector Phidalitz(part->Px(), part->Py(), part->Pz(), part->Energy());
-  Decay(333, &Phidalitz);
+  //Decay(333, &Phidalitz, dp3[0]->GetPdgCode());
```
remove

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -927,6 +990,7 @@ void AliDecayerExodus::Decay(TClonesArray *array)
   if( dp2[0]->GetPdgCode() != 11 && (TMath::Abs(dp2[1]->GetPdgCode()) != 11))   continue;
 
   TLorentzVector Phi(part->Px(), part->Py(), part->Pz(), part->Energy());
+  //Decay(333, &Phi, 0);
```
remove

qgp commented on EVGEN/AliDecayerExodus.cxx:
```diff
@@ -949,6 +1013,7 @@ void AliDecayerExodus::Decay(TClonesArray *array)
   if((dp2[0]->GetPdgCode() != 11) && ((TMath::Abs(dp2[1]->GetPdgCode()) != 11)))   continue;
 
   TLorentzVector JPsi(part->Px(), part->Py(), part->Pz(), part->Energy());
+  //Decay(443, &JPsi, 0);
```
remove

qgp commented on EVGEN/AliDecayerExodus.h:
```diff
@@ -35,6 +35,7 @@ class AliDecayerExodus : public AliDecayer
  public:
     AliDecayerExodus();
     virtual void    Init();
+  //  virtual void    Decay(Int_t idpart,TLorentzVector* pparent, Int_t idpartner);
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1588,7 +1588,8 @@ void AliPythia::PizeroDalitz()
   TLorentzVector pizero(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
-  fExodus.Decay(pdg, &pizero);
+  //fExodus.Decay(pdg, &pizero, fPyjets->K[1][fd]);
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1613,7 +1614,8 @@ void AliPythia::EtaDalitz()
   TLorentzVector eta(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
-  fExodus.Decay(pdg, &eta);
+  //fExodus.Decay(pdg, &eta,fPyjets->K[1][fd] );
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1638,6 +1640,7 @@ void AliPythia::RhoDirect()
   continue;
   TLorentzVector rho(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
+  //fExodus.Decay(pdg, &rho, 0);
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1663,7 +1666,8 @@ void AliPythia::OmegaDalitz()
   TLorentzVector omegadalitz(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
-  fExodus.Decay(pdg, &omegadalitz);
+  //fExodus.Decay(pdg, &omegadalitz, fPyjets->K[1][fd]);
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1688,6 +1692,7 @@ void AliPythia::OmegaDirect()
   continue;
   TLorentzVector omegadirect(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
+  //fExodus.Decay(pdg, &omegadirect, 0);
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1709,11 +1714,12 @@ void AliPythia::EtaprimeDalitz()
   TLorentzVector etaprime(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
-  fExodus.Decay(pdg, &etaprime);
+  //fExodus.Decay(pdg, &etaprime, fPyjets->K[1][fd]);
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1734,11 +1740,12 @@ void AliPythia::PhiDalitz()
   TLorentzVector phidalitz(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
-  fExodus.Decay(pdg, &phidalitz);
+  //fExodus.Decay(pdg, &phidalitz, fPyjets->K[1][fd]);
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1763,6 +1770,7 @@ void AliPythia::PhiDirect()
   continue;
   TLorentzVector phi(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
+  //fExodus.Decay(pdg, &phi, 0);
```
remove

qgp commented on PYTHIA6/AliPythia6/AliPythia.cxx:
```diff
@@ -1787,6 +1795,7 @@ void AliPythia::JPsiDirect()
   continue;
   TLorentzVector jpsi(fPyjets->P[0][i], fPyjets->P[1][i], fPyjets->P[2][i], fPyjets->P[3][i]);
   Int_t pdg = TMath::Abs(fPyjets->K[1][i]);
+  //fExodus.Decay(pdg, &jpsi, 0);
```
remove

otonvd commented:
Dear all,
I cleaned the code as suggested. With respect to a more detailed description of this code and its possible implications for any pythia simulation:

- All changes in the code are affecting exclusively LMee simulations with the Exodus decayer.
- For the Pythia part of the code, still only affecting the Exodus decayer, in particular what is done here is to accommodate the code for handling two new decays added in the decay table to dielectrons (namely eta' -> omega e+ e- and phi -> pi0 e+ e-).

Now one question regarding the pull request. Since the time when it was first proposed I changed my AliRoot installations, folders, etc. Now I have an updated AliRoot and the changes with respect to the master are still only this 5 files that I want to pull, but I don't know if I do it right now it will be part of this old pull request or it will be done as a new one. Any help on this?

Cheers
oton

otonvd commented:
Not having received any suggestion on how to do it, I cancel this pull request and create a new one with the same files.

miweberSMI commented:
Hi @otonvd,  

can you mention the new pull request here?

Thanks 
Michael

otonvd commented:
New pull request is: #549 
https://github.com/alisw/AliRoot/pull/549
